octokit/resources.py

```diff
@@ -118,6 +125,41 @@ def parse_schema_list(self, data, name):
 
     return schema
 
+  # Parse pagination links from the headers
+  def process_rels(self, headers):
+    if not 'Link' in headers:
+      return {}
+
+    regex = r'<(?P<href>.*?)>; rel="(?P<name>\w+)"'
+    rels = headers['Link'].split(', ')
+    def createResource(rel):
+      (name, href) = re.search(regex, rel).group('name', 'href')
+      return (name, Resource(self.session, url=href))
+
+    return dict(map(createResource, rels))
+
+  # Continue following the relations until there are no more links
+  #
+  # *args           - Uri template argument
+  # **kwargs        – Uri template arguments
+  # Returns a list of resources
+  def paginate(self, *args, **kwargs):
+    if self.auto_paginate or self.per_page:
```

Comment:
You can simplify this to

    if 'per_page' not in kwargs:
        kwargs['per_page'] = 100 if self.auto_paginate else self.per_page